lib/rubygems/test_case.rb

```diff
@@ -222,17 +224,23 @@ def setup
     @orig_gem_vendor = ENV['GEM_VENDOR']
     @orig_gem_spec_cache = ENV['GEM_SPEC_CACHE']
     @orig_rubygems_gemdeps = ENV['RUBYGEMS_GEMDEPS']
+    @orig_bundle_gemfile   = ENV['BUNDLE_GEMFILE']
     @orig_rubygems_host = ENV['RUBYGEMS_HOST']
+    @orig_bundle_disable_postit = ENV['BUNDLE_DISABLE_POSTIT']
     ENV.keys.find_all { |k| k.start_with?('GEM_REQUIREMENT_') }.each do |k|
       ENV.delete k
     end
     @orig_gem_env_requirements = ENV.to_hash
 
     ENV['GEM_VENDOR'] = nil
+    ENV['BUNDLE_DISABLE_POSTIT'] = 'true'
 
     @current_dir = Dir.pwd
     @fetcher     = nil
-    @ui          = Gem::MockGemUi.new
+
+    Bundler.ui                     = Bundler::UI::Silent.new
+    @ui                            = Gem::MockGemUi.new
+    Gem::DefaultUserInteraction.ui = Gem::MockGemUi.new
```

Comment:
no, because we explicitly call `use_ui @ui` when we want to capture output in the tests. changing this breaks a few tests